### ISSUE TITLE
Error in eigen(delta1) : infinite or missing values in 'x' if taxa_are_rows = T

### ISSUE THREAD

**Issue by manasishah86** [joey711/phyloseq#754]
Hi Joey,

I am using phyloseq: 
`> packageVersion("phyloseq")
[1] ‘1.20.0’`

I have a presence-absence normalized OTU table (attached). If I set taxa_are_rows = F, I can run ordinate as follows:
`ordPresAbs = ordinate(otu_table_ps, method="PCoA", distance="jaccard")`

However, if I set taxa_are_rows = T, while generating the ps object I get the following error with the same command:
`ordPresAbs = ordinate(otu_table_ps, method="PCoA", distance="jaccard")`
`Error in eigen(delta1) : infinite or missing values in 'x'`

The taxa_are_rows = T flag is required for me to make a meaningful phyloseq object with a taxonomy table. This is the  first time I am getting such an error. Please find the problematic ps object attached which had the `taxa_are_rows = T` attached as otu_table_ps and the non problematic object with `taxa_are_rows=F` attached as otu_table_F

tl:dr : How to correct ordinate function error when taxa_are_rows = T while making ps object

Thanks for your help!

[sample_data.zip](https://github.com/joey711/phyloseq/files/987826/sample_data.zip)

**Comment by manasishah86** [comment]
Fixed it. The issue was I had some samples with zero total abundance and that was causing the Jaccard distance calculation to break.

**Comment by joey711** [comment]
`taxa_are_rows` is just a descriptor of the abundance table. It can be either TRUE or FALSE and a taxonomy table can still be included. They are unrelated.

Glad you were able to fix the Jaccard issue

**Comment by Marielis24** [comment]
> Fixed it. The issue was I had some samples with zero total abundance and that was causing the Jaccard distance calculation to break.

Hello, how did you solve this? I am having the same problem.

**Comment by manasishah86** [comment]
Hi, you have to filter out the zero abundance samples

Best,
Manasi

**Comment by studsarah** [comment]
Hello,
I have the same problem but with "wunifrac". I already filtered zero abundance samples but still have this error ... Any help please ?

**Comment by fconstancias** [comment]
I was able to solve it using : 
```
tsne_out <- Rtsne::Rtsne(iDist1, dims = 3, perplexity = 20, verbose = T) 
proj <- tsne_out$Y %>% as.data.frame()
rownames(proj) <- rownames(t(otu_table(physeq)))
plot_ordination(physeq, proj)
```

**Comment by FlorenciaTL** [comment]
Hi,
I have the same issue, how do you filter out the zero abundance samples?

Thanks
Flor

**Comment by Samandthebees** [comment]
I am also having the same issue, none of my rows have 0 abundance and I am very confused as to why this is not working
